Tests/SwaggerParserTests/SwaggerParserTests.swift

```diff
@@ -1,20 +1,159 @@
+                guard case .object(let baseSchema) = structure.structure else {
+                    throw GetBaseAndChildSchemasError.badSubschemaType(subschema)
+                }
+                base = baseSchema
+                
+            default:
+                throw GetBaseAndChildSchemasError.badSubschemaType(subschema)
+            }
         }
     }
+    
+    guard base != nil else {
+        throw GetBaseAndChildSchemasError.missingBase
+    }
+    
+    guard child != nil else {
+        throw GetBaseAndChildSchemasError.missingChild
+    }
+    
+    return (base: base, child: child)
+}
+
+/// MARK: Validation functions
+
+fileprivate func validate(testAllOfBaseSchema schema: ObjectSchema) {
+    validate(thatSchema: schema, named: "TestAllOfBase", hasRequiredProperties: ["base", "test_type"])
+}
+
+fileprivate func validate(thatSchema schema: ObjectSchema, named name: String, hasRequiredProperties properties: [String]) {
+    XCTAssertEqual(schema.properties.count, properties.count)
+    XCTAssertEqual(schema.required, properties)
+    
+    properties.forEach { property in
+        XCTAssertNotNil(schema.properties.first(where: {$0.key == property}))
+    }
+}
+
+fileprivate func validate(thatParameter parameter: Parameter, named parameterName: String, isAnObjectNamed objectName: String, withPropertyName objectPropertyName: String) {
+    guard case .body(_, let schema) = parameter else {
+        return XCTFail("\(parameterName) is not a .body.")
+    }
+    
+    guard case .structure(let structure) = schema else {
+        return XCTFail("\(parameterName)'s schema is not a .structure.")
+    }
+    
+    XCTAssertEqual(structure.name, objectName)
+    
+    guard case .object(let object) = structure.structure else {
+        return XCTFail("\(parameterName)'s schema's structure is not an .object.")
+    }
+    
+    XCTAssertTrue(object.properties.contains(where: {$0.key == objectPropertyName}))
+}
+
+fileprivate func validate(thatChildSchema childSchema: Schema, named childName: String, withProperties childProperties: [String], hasParentNamed parentName: String, withProperties parentProperties: [String]) {
+    guard case .allOf(let childAllOf) = childSchema else {
+        XCTFail("\(childName) is not an allOf."); return
+    }
+    
+    XCTAssertEqual(childAllOf.subschemas.count, 2)
+    
+    guard
+        let childsParent = childAllOf.subschemas.first,
+        case .structure(let childsParentStructure) = childsParent,
+        childsParentStructure.name == parentName,
+        case .object(let childsParentSchema) = childsParentStructure.structure else
+    {
+        return XCTFail("\(childName)'s parent is not a Structure<Schema.object>")
+    }
+    
+    validate(thatSchema: childsParentSchema, named: parentName, hasRequiredProperties: parentProperties)
+    
+    guard
+        let child = childAllOf.subschemas.last,
+        case .object(let childSchema) = child else
```

Comment:
Can't you fit this all onto one line? Usually I set my guide at 110 characters (that's about the maximum Github can display cleanly in the diff view).
```
guard let child = childAllOf.subschemas.last, case .object(let childSchema) = child else {
```